File: app/src/main/java/tech/inscripts/ins_armman/mMitra/utility/Utility.java
package tech.inscripts.ins_armman.mMitra.utility;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.res.Resources;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.LayerDrawable;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Base64;
import android.util.DisplayMetrics;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;
import tech.inscripts.ins_armman.mMitra.R;
import tech.inscripts.ins_armman.mMitra.data.database.DatabaseContract;
import tech.inscripts.ins_armman.mMitra.data.database.DatabaseManager;
import tech.inscripts.ins_armman.mMitra.data.GraphPoint;
import org.joda.time.*;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.lang.reflect.Type;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

import static tech.inscripts.ins_armman.mMitra.utility.Constants.*;
import static tech.inscripts.ins_armman.mMitra.utility.Keywords.*;


/**
 * Created by Amol on 10/10/17.
 */

public class Utility {
    public static String TAG = Utility.class.getName();
    public static final String COMMAN_PREF_NAME="CommonPrefs";
    public static final String LANGUAGE = "LANGUAGE";
    public static final SimpleDateFormat mDateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.US);
    public static final SimpleDateFormat mDateTimeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US);
    public static final SimpleDateFormat mDateDisplayFormat = new SimpleDateFormat("dd-MM-yyyy");

    /**
     * change the locale for the app.
     *
     * @param context
     * @param locale
     */
    public static void setApplicationLocale(Context context, String locale) {
        try {
            setLocaleInPreference(locale, context);

            Resources res = context.getApplicationContext().getResources();
            // Change locale settings in the app.
            DisplayMetrics dm = res.getDisplayMetrics();
            android.content.res.Configuration conf = res.getConfiguration();
            String[] localeArray = locale.split("_");
            if (localeArray.length > 1) {
                conf.locale = new Locale(localeArray[0], localeArray[1]);
            } else {
                conf.locale = new Locale(locale);
            }
            res.updateConfiguration(conf, dm);
        } catch (Exception ex) {
        }
    }

    private static void setLocaleInPreference(String locale, Context context) {
        SharedPreferences prefs = context.getSharedPreferences(COMMAN_PREF_NAME, Activity.MODE_PRIVATE);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(LANGUAGE, locale);
        editor.commit();
    }

    public static String getLanguagePreferance(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(COMMAN_PREF_NAME, Activity.MODE_PRIVATE);
        String language = prefs.getString(LANGUAGE, "");
        return language;
    }

    /**
     * @param view         View to animate
     * @param toVisibility Visibility at the end of animation
     * @param toAlpha      Alpha at the end of animation
     * @param duration     Animation duration in ms
     */
    public static void animateView(final View view, final int toVisibility, float toAlpha, int duration) {
        boolean show = toVisibility == View.VISIBLE;
        if (show) {
            view.setAlpha(0);
        }
        view.setVisibility(View.VISIBLE);
        view.animate()
                .setDuration(duration)
                .alpha(show ? toAlpha : 0)
                .setListener(new AnimatorListenerAdapter() {
                    @Override
                    public void onAnimationEnd(Animator animation) {
                        view.setVisibility(toVisibility);
                    }
                });
    }

    public static boolean hasInternetConnectivity(Context context) {
        boolean rc = false;
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        if (cm != null) {
            try {
                NetworkInfo activeNetworkInfo = cm.getActiveNetworkInfo();
                if (activeNetworkInfo != null && activeNetworkInfo.isAvailable() && activeNetworkInfo.isConnected()) {
                    rc = true;
                }
            } catch (Exception e) {
            }
        }
        return rc;
    }

    public static ArrayList<String> getDeviceImeiNumber(Context context) {
        TelephonyInfo telephonyInfo = TelephonyInfo.getInstance(context);
        ArrayList<String> imeiArray = new ArrayList<>();
//        imeiArray.add("867375022232910");
        if (telephonyInfo.isDualSIM()) {
            imeiArray.add(telephonyInfo.getImsiSIM1());
            imeiArray.add(telephonyInfo.getImsiSIM2());
        } else {
            imeiArray.add(telephonyInfo.getImsiSIM1());
        }

        return imeiArray;
    }

    public static SQLiteDatabase getDatabase() {
        return DatabaseManager.getInstance().openDatabase();
    }

    public static String generateUniqueId() {
        return System.currentTimeMillis() + UNIQUE_MEMBER_ID_SEPERATOR + getUserId();
    }

    public static String getUserId() {
       Cursor cursor = getDatabase().rawQuery("SELECT " + DatabaseContract.LoginTable.COLUMN_USER_ID
        + " FROM " + DatabaseContract.LoginTable.TABLE_NAME, null);

        if (cursor.moveToFirst()) return cursor.getString(cursor.getColumnIndex(DatabaseContract.LoginTable.COLUMN_USER_ID));
        else return "0";
    }

    /**
     * This is the method which generates the Hash.
     * @param s = JsonResponse
     * @return hash in string format
     */
    public static String mdFive(final String s) {
        final String MD5 = "MD5";
        try {
            // Create MD5 Hash
            MessageDigest digest = MessageDigest
                    .getInstance(MD5);
            digest.update(s.getBytes());
            byte messageDigest[] = digest.digest();

            // Create Hex String
            StringBuilder hexString = new StringBuilder();
            for (byte aMessageDigest : messageDigest) {
                String h = Integer.toHexString(0xFF & aMessageDigest);
                while (h.length() < 2)
                    h = "0" + h;
                hexString.append(h);
            }
            return hexString.toString();

        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }
        return "";
    }

    public static String getStringFromBitmap(Bitmap bitmapPicture) {
        return Base64.encodeToString(getImageByteArray(bitmapPicture), Base64.DEFAULT);
    }

    public static byte[] getImageByteArray(Bitmap bitmap){
        final int COMPRESSION_QUALITY = 100;
        ByteArrayOutputStream byteArrayBitmapStream = new ByteArrayOutputStream();
        bitmap.compress(Bitmap.CompressFormat.PNG, COMPRESSION_QUALITY,
                byteArrayBitmapStream);
        return byteArrayBitmapStream.toByteArray();
    }

    public static void setBadgeCount(Context context, LayerDrawable icon, int count) {

        BadgeDrawable badge;

        // Reuse drawable if possible
        Drawable reuse = icon.findDrawableByLayerId(
                R.id.ic_badge);
        if (reuse != null && reuse instanceof BadgeDrawable) {
            badge = (BadgeDrawable) reuse;
        } else {
            badge = new BadgeDrawable(context);
        }

        badge.setCount(count);
        icon.mutate();
        icon.setDrawableByLayerId(R.id.ic_badge, badge);
    }

    public static String getCurrentDateTime() {
        return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US).format(new Date());
    }

    public static int getDaysBetweenTwoDates(String dateOne, String dateTwo) {
        return Days.daysBetween(new DateTime(dateOne), new DateTime(dateTwo)).getDays();
    }

    public static String getMonthForInt(int num) {
        Calendar cal = Calendar.getInstance();
        cal.set(Calendar.MONTH, num);
        SimpleDateFormat month_date = new SimpleDateFormat("MMM", Locale.US);
        String month_name = month_date.format(cal.getTime());
        return month_name;
    }

    public static String getSamMamUsingMUAC(float muac, Context context) {
        String samMamStatus = "";
        if (muac != -1 && muac <= 11.5) {
            samMamStatus = context.getString(R.string.sam_status);
        } else if (muac > 11.5 && muac <= 12.5) {
            samMamStatus = context.getString(R.string.mam_status);
        }
        return samMamStatus;
    }

    public static int getAgeInMonths(int year, int month, int day) {
        LocalDate birthdate = new LocalDate(year, month + 1, day);          //Birth date
        LocalDate now = new LocalDate();                    //Today's date
        Period period = new Period(birthdate, now, PeriodType.months());
        int months = period.getMonths();
        return months;
    }

    public static int getAgeInMonths(String dob) {
        int year = DateUtility.getYearFromDate(dob);
        int month = DateUtility.getMonthFromDate(dob);
        int day = DateUtility.getDayFromDate(dob);
        int ageInMonths = getAgeInMonths(year, month, day);
        return ageInMonths;
    }

    public static String getWeightCategory(float weight, GraphPoint graphPoint) {

        String weightType = WEIGHT_TYPE_NORMAL_WEIGHT;

        Float median = graphPoint.getMedian();
        Float _2SD = graphPoint.get_2SD();
        Float _3SD = graphPoint.get_3SD();

        if (weight > median) {
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_HIGH_WEIGHT);
            weightType = WEIGHT_TYPE_HIGH_WEIGHT;
        } else if (weight >= _2SD && weight <= median) {
            weightType = WEIGHT_TYPE_NORMAL_WEIGHT;
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_NORMAL_WEIGHT);
        } else if (weight >= _3SD && weight < _2SD) {
            weightType = WEIGHT_TYPE_LOW_WEIGHT;
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_LOW_WEIGHT);
        } else if (weight < _3SD) {
            weightType = WEIGHT_TYPE_SEVERELY_LOW_WEIGHT;
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_SEVERELY_LOW_WEIGHT);
        }

        return weightType;
    }

    public static ArrayList<GraphPoint> getStaticZScorePoints(String gender, Context context) {
        if(gender.equalsIgnoreCase(null) || gender.equalsIgnoreCase(""))
        {
            gender = "M";
        }
        String fileName = null;
        if (gender.equalsIgnoreCase(MALE) || gender.equalsIgnoreCase("M") || gender.equalsIgnoreCase(GENDER_MALE)) {
            fileName = "wfa_boys_0_5_zscores.json";
        } else {
            fileName = "wfa_girls_0_5_zscores.json";
        }

        String jsonString = Utility.loadJSONFromAsset(context, fileName);
        Type typeOfObjectsList = new TypeToken<ArrayList<GraphPoint>>() {
        }.getType();
        ArrayList<GraphPoint> graphPointArrayList = new GsonBuilder().create().fromJson(jsonString, typeOfObjectsList);
        return graphPointArrayList;
    }

    public static String loadJSONFromAsset(Context context, String jsonFileName) {
        String json = null;
        try {
            InputStream is = context.getAssets().open(jsonFileName);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
        } catch (IOException ex) {
            ex.printStackTrace();
            return null;
        }
        return json;
    }

    public static int getIntForMonth(String monthName, Locale locale) {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("MMM", locale);
        try {
            Date date = simpleDateFormat.parse(monthName);
            Calendar cal = Calendar.getInstance();
            cal.setTime(date);
            int month = cal.get(Calendar.MONTH);
            return month;
        } catch (ParseException e) {
            e.printStackTrace();
        }

        return -1;
    }
    public static void hideKeyboard(Context context, View view) {
        InputMethodManager inputMethodManager = (InputMethodManager) context.getSystemService(Activity.INPUT_METHOD_SERVICE);
        inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
    }

    public static String getVillageName(String villageId) {
        Cursor cursor = getDatabase().rawQuery("SELECT * FROM "
                + DatabaseContract.VillageTable.TABLE_NAME
                + " WHERE "
                + DatabaseContract.VillageTable.COLUMN_VILLAGE_ID + " = ? ", new String[]{villageId});

        if(cursor.moveToFirst()) return cursor.getString(cursor.getColumnIndex(DatabaseContract.VillageTable.COLUMN_VILLAGE_NAME));
        return villageId;
    }

    public static String getAppVersionName(Context context) {
        PackageInfo pInfo = null;
        try {
            pInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }
        return pInfo.versionName;
    }
}

File: app/src/main/java/tech/inscripts/ins_armman/mMitra/utility/Utility.kt
package tech.inscripts.ins_armman.mMitra.utility

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.app.Activity
import android.content.Context
import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import android.content.res.Configuration
import android.content.res.Resources
import android.database.sqlite.SQLiteDatabase
import android.graphics.Bitmap
import android.graphics.drawable.LayerDrawable
import android.net.ConnectivityManager
import android.util.Base64
import android.util.DisplayMetrics
import android.view.View
import android.view.inputmethod.InputMethodManager
import org.joda.time.*
import tech.inscripts.ins_armman.mMitra.R
import tech.inscripts.ins_armman.mMitra.data.GraphPoint
import tech.inscripts.ins_armman.mMitra.data.database.DatabaseContract
import tech.inscripts.ins_armman.mMitra.data.database.DatabaseManager
import tech.inscripts.ins_armman.mMitra.utility.Constants.*
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.nio.charset.Charset
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class Utility {
    var dateUtility = DateUtility()
    val obj_telephonyInfo = TelephonyInfo()
    var TAG = Utility::class.java.name
    val COMMAN_PREF_NAME = "CommonPrefs"
    val Language = "LANGUAGE"
    val mDateFormat: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.US)
    val mDateTimeFormat: SimpleDateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US)
    val mDateDisplayFormat: SimpleDateFormat = SimpleDateFormat("dd-MM-yyyy")


    fun setLocaleInPreference(locale: String, context: Context) {
        val prefs = context.getSharedPreferences(COMMAN_PREF_NAME, Activity.MODE_PRIVATE)
        val editor = prefs.edit()
        editor.putString(Language, locale)
        editor.commit()
    }

    fun getLanguagePreferance(context: Context?): String {
        val prefs = context!!.getSharedPreferences(COMMAN_PREF_NAME, Activity.MODE_PRIVATE)
        return prefs.getString(Language, "")
    }

    /**
     * change the language for the app.
     *
     * @param context
     * @param locale
     */
    fun setApplicationLocale(context: Context, locale: String) {
        try {
            setLocaleInPreference(locale, context)

            var res: Resources = context.applicationContext.resources
            // Change locale settings in the app.
            var dm: DisplayMetrics = res.displayMetrics
            val conf: Configuration = res.configuration
            val localeArray = locale.split("_".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            if (localeArray.size > 1) {
                conf.locale = Locale(localeArray[0], localeArray[1])
            } else {
                conf.locale = Locale(locale)
            }
            res.updateConfiguration(conf, dm)
        } catch (ex: Exception) {
        }

    }


    /**
     * @param view         View to animate
     * @param toVisibility Visibility at the end of animation
     * @param toAlpha      Alpha at the end of animation
     * @param duration     Animation duration in ms
     */
    fun animateView(view: View, toVisibility: Int, toAlpha: Float, duration: Int) {
        val show = toVisibility == View.VISIBLE
        if (show) {
            view.alpha = 0f
        }
        view.visibility = View.VISIBLE
        view.animate()
            .setDuration(duration.toLong())
            .alpha(if (show) toAlpha else 0F)
            .setListener(object : AnimatorListenerAdapter() {
                override fun onAnimationEnd(animation: Animator) {
                    view.visibility = toVisibility
                }
            })
    }

    fun hasInternetConnectivity(context: Context?): Boolean {
        var rc = false
        val cm = context?.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
        if (cm != null) {
            try {
                val activeNetworkInfo = cm.activeNetworkInfo
                if (activeNetworkInfo != null && activeNetworkInfo.isAvailable && activeNetworkInfo.isConnected) {
                    rc = true
                }
            } catch (e: Exception) {
            }

        }
        return rc
    }

    fun getDeviceImeiNumber(context: Context): ArrayList<String> {
        val telephonyInfo = obj_telephonyInfo.getInstance(context)
        val imeiArray = ArrayList<String>()
               imeiArray.add("869432026925037");
       /* if (telephonyInfo.isDualSIM()) {
            imeiArray.add(telephonyInfo.imsiSIM1)
            imeiArray.add(telephonyInfo.imsiSIM2)
        } else {
            imeiArray.add(telephonyInfo.imsiSIM1)
        }*/

        return imeiArray
    }

    fun getDatabase(): SQLiteDatabase {
        return DatabaseManager.getInstance().openDatabase()
    }

    fun generateUniqueId(): String {
        var a=System.currentTimeMillis()
        var b=UNIQUE_MEMBER_ID_SEPERATOR
        var c=getUserId()
        var d: String=""
        if(true)
            d=a.toString() +b +c
        return d
    }

    fun getUserId(): String {
        val cursor = getDatabase().rawQuery(
            "SELECT " + DatabaseContract.LoginTable.COLUMN_USER_ID
                    + " FROM " + DatabaseContract.LoginTable.TABLE_NAME, null
        )

        return if (cursor.moveToFirst())
            cursor.getString(cursor.getColumnIndex(DatabaseContract.LoginTable.COLUMN_USER_ID))
        else
            "0"
    }

    /**
     * This is the method which generates the Hash.
     * @param s = JsonResponse
     * @return hash in string format
     */
    fun mdFive(s: String): String {
        val MD5 = "MD5"
        try {
            // Create MD5 Hash
            val digest = MessageDigest
                .getInstance(MD5)
            digest.update(s.toByteArray())
            val messageDigest = digest.digest()

            // Create Hex String
            val hexString = StringBuilder()
            for (aMessageDigest in messageDigest) {
                var h = Integer.toHexString(0xFF and aMessageDigest.toInt())
                while (h.length < 2)
                    h = "0$h"
                hexString.append(h)
            }
            return hexString.toString()

        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }

        return ""
    }

    fun getStringFromBitmap(bitmapPicture: Bitmap): String {
        return Base64.encodeToString(getImageByteArray(bitmapPicture), Base64.DEFAULT)
    }

    fun getImageByteArray(bitmap: Bitmap): ByteArray {
        val COMPRESSION_QUALITY = 100
        val byteArrayBitmapStream = ByteArrayOutputStream()
        bitmap.compress(
            Bitmap.CompressFormat.PNG, COMPRESSION_QUALITY,
            byteArrayBitmapStream
        )
        return byteArrayBitmapStream.toByteArray()
    }

    fun setBadgeCount(context: Context, icon: LayerDrawable, count: Int) {

        val badge: BadgeDrawable

        // Reuse drawable if possible
        val reuse = icon.findDrawableByLayerId(
            R.id.ic_badge
        )
        if (reuse != null && reuse is BadgeDrawable) {
            badge = reuse
        } else {
            badge = BadgeDrawable(context)
        }

        badge.setCount(count)
        icon.mutate()
        icon.setDrawableByLayerId(R.id.ic_badge, badge)
    }

    fun getCurrentDateTime(): String {
        return SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US).format(Date())
    }

    fun getDaysBetweenTwoDates(dateOne: String, dateTwo: String): Int {
        return Days.daysBetween(DateTime(dateOne), DateTime(dateTwo)).days
    }

    fun getMonthForInt(num: Int): String {
        val cal = Calendar.getInstance()
        cal.set(Calendar.MONTH, num)
        val month_date = SimpleDateFormat("MMM", Locale.US)
        return month_date.format(cal.time)
    }

    fun getSamMamUsingMUAC(muac: Float, context: Context): String {
        var samMamStatus = ""
        if (muac != -1f && muac <= 11.5) {
            samMamStatus = context.getString(R.string.sam_status)
        } else if (muac > 11.5 && muac <= 12.5) {
            samMamStatus = context.getString(R.string.mam_status)
        }
        return samMamStatus
    }

    fun getAgeInMonths(year: Int, month: Int, day: Int): Int {
        val birthdate = LocalDate(year, month + 1, day)          //Birth date
        val now = LocalDate()                    //Today's date
        val period = Period(birthdate, now, PeriodType.months())
        return period.months
    }

    fun getAgeInMonths(dob: String): Int {
        val year = dateUtility.getYearFromDate(dob)
        val month = dateUtility.getMonthFromDate(dob)
        val day = dateUtility.getDayFromDate(dob)
        return getAgeInMonths(year, month, day)
    }

    fun getWeightCategory(weight: Float, graphPoint: GraphPoint): String {

        var weightType = WEIGHT_TYPE_NORMAL_WEIGHT

        val median = graphPoint.getMedian()
        val _2SD = graphPoint.get_2SD()
        val _3SD = graphPoint.get_3SD()

        if (weight > median) {
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_HIGH_WEIGHT);
            weightType = WEIGHT_TYPE_HIGH_WEIGHT
        } else if (weight >= _2SD && weight <= median) {
            weightType = WEIGHT_TYPE_NORMAL_WEIGHT
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_NORMAL_WEIGHT);
        } else if (weight >= _3SD && weight < _2SD) {
            weightType = WEIGHT_TYPE_LOW_WEIGHT
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_LOW_WEIGHT);
        } else if (weight < _3SD) {
            weightType = WEIGHT_TYPE_SEVERELY_LOW_WEIGHT
            //Log.d("****", "Weight:"+weight + "WeightCategory:"+AWWConstants.WEIGHT_TYPE_SEVERELY_LOW_WEIGHT);
        }

        return weightType
    }

    /**
     * this method is for child
     */
    /*fun getStaticZScorePoints(gender: String, context: Context): ArrayList<GraphPoint>? {
        var gender = gender
        if (gender.equals(null!!, ignoreCase = true) || gender.equals("", ignoreCase = true)) {
            gender = "M"
        }
        var fileName: String? = null
        if (gender.equals(MALE, ignoreCase = true) || gender.equals(
                "M",
                ignoreCase = true
            ) || gender.equals(GENDER_MALE, ignoreCase = true)
        ) {
            fileName = "wfa_boys_0_5_zscores.json"
        } else {
            fileName = "wfa_girls_0_5_zscores.json"
        }

        val jsonString =loadJSONFromAsset(context, fileName)
        val typeOfObjectsList = object : TypeToken<ArrayList<GraphPoint>>() {

        }.type
        return GsonBuilder().create().fromJson<ArrayList<GraphPoint>>(jsonString, typeOfObjectsList)
    }
*/
    fun loadJSONFromAsset(context: Context, jsonFileName: String): String? {
        var json: String? = null
        try {
            val inputStream = context.assets.open(jsonFileName)
            val size = inputStream.available()
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()
            json = String(buffer, Charset.forName("UTF-8"))
        } catch (ex: IOException) {
            ex.printStackTrace()
            return null
        }

        return json
    }

    fun getIntForMonth(monthName: String, locale: Locale): Int {
        val simpleDateFormat = SimpleDateFormat("MMM", locale)
        try {
            val date = simpleDateFormat.parse(monthName)
            val cal = Calendar.getInstance()
            cal.time = date
            return cal.get(Calendar.MONTH)
        } catch (e: ParseException) {
            e.printStackTrace()
        }

        return -1
    }

    fun hideKeyboard(context: Context, view: View) {
        val inputMethodManager = context.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
    }

    fun getVillageName(villageId: String): String {
        val cursor = getDatabase().rawQuery(
            "SELECT * FROM "
                    + DatabaseContract.VillageTable.TABLE_NAME
                    + " WHERE "
                    + DatabaseContract.VillageTable.COLUMN_VILLAGE_ID + " = ? ", arrayOf(villageId)
        )

        return if (cursor.moveToFirst()) cursor.getString(cursor.getColumnIndex(DatabaseContract.VillageTable.COLUMN_VILLAGE_NAME)) else villageId
    }

    fun getAppVersionName(context: Context): String {
        var pInfo: PackageInfo? = null
        try {
            pInfo = context.packageManager.getPackageInfo(context.packageName, 0)
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }

        return pInfo!!.versionName
    }
}